latch/types/metadata.py

```diff
@@ -57,6 +57,71 @@ class LatchAuthor:
     github: Optional[str] = None
 
 
+@dataclass(frozen=True)
+class FlowBase:
+    ...
+
+
+@dataclass(frozen=True, init=False)
+class Section(FlowBase):
+    section: str
+    flow: List[FlowBase]
+
+    def __init__(self, section: str, *flow: FlowBase):
+        object.__setattr__(self, "section", section)
+        object.__setattr__(self, "flow", list(flow))
+
+
+@dataclass(frozen=True)
+class Text(FlowBase):
+    text: str
+
+
+@dataclass(frozen=True)
+class Title(FlowBase):
+    title: str
+
+
+@dataclass(frozen=True, init=False)
+class Params(FlowBase):
+    params: List[str]
+
+    def __init__(self, *args: str):
+        object.__setattr__(self, "params", list(args))
+
+
+@dataclass(frozen=True, init=False)
+class Spoiler(FlowBase):
+    spoiler: str
+    flow: List[FlowBase]
+
+    def __init__(self, spoiler: str, *flow: FlowBase):
+        object.__setattr__(self, "spoiler", spoiler)
+        object.__setattr__(self, "flow", list(flow))
+
+
+@dataclass(frozen=True, init=False)
+class ForkBranch:
+    display_name: str
+    flow: List[FlowBase]
+
+    def __init__(self, display_name: str, *flow: FlowBase):
+        object.__setattr__(self, "display_name", display_name)
+        object.__setattr__(self, "flow", list(flow))
+
+
+@dataclass(frozen=True, init=False)
+class Fork(FlowBase):
+    fork: str
+    display_name: str
+    flows: Dict[str, ForkBranch]
+
+    def __init__(self, fork: str, display_name: str, /, **flows: ForkBranch):
```

Comment:
whats that weird slash

Reply:
positional argument delimiter, but its not supported in 3.7 so ill remove it